### DISCUSSION TITLE
Option to only report on file changes?

### DISCUSSION THREAD

**Discussion by rsoesemann** [pmd/pmd-github-action#5] [Ideas]
Does this offical Github Action have an option to run PMD on diffs only? Like the one in https://github.com/marketplace/actions/pmd-analyser? Or do you plan to add such a helpful option?

**Comment by adangel** [comment]
No, not yet. I kept it simple for the beginning.
As this is an open source project, pull requests are always welcome.

Note: analyzing only the changed file might work with Apex, but maybe not for Java - the changed file might affect other files (which depend on the changed files), so to be correct, one would need to analyze the changed files + any file, that uses the changed file. This is not a problem yet for Apex, since we only look at the source code and don't resolve any types.
